### PR TITLE
feat(containers/selectobject): allow to filter on everything

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

We use this container in TDP for the copy/move feature, so we need to allow the user to search for any folder in any sub folder.

**What is the chosen solution to this problem?**

Introduce a new `filterMode` property to allow to filter the whole folder tree.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/SelectObject/SelectObject.container.js [hunk 1]
```diff
@@ -88,6 +92,38 @@ export function filter(
...
+			const children = item.get('children', null);
+			if (name.toLowerCase().includes(query.toLowerCase())) {
+				onMatch(item);
+				results = results.push(item.set('currentPosition', currentPosition));
+			}
+			if (children) {
+				results = results.concat(
+					filterAll(children, query, { nameAttr }, `${currentPosition} > ${name}`),
+				);
+			}
```
Comment (nicomaligne) [inline packages/containers/src/SelectObject/SelectObject.container.js]: Can you cut this block  to be more readable ? More functionnal oriented ? What do u think about it ?

File: packages/containers/src/SelectObject/SelectObject.container.js [hunk 2]
```diff
@@ -88,6 +92,38 @@ export function filter(
...
+		});
+	}
+
+	return results;
```
[the remark this answers is not in the document]
Comment (ncomont) [reply packages/containers/src/SelectObject/SelectObject.container.js]: I don't see any difference. The `result` list must exists to be returned if a query is present. Maybe I'm missing something ?

File: packages/containers/src/SelectObject/SelectObject.container.js (hunk 1 above)
Comment (ncomont) [reply packages/containers/src/SelectObject/SelectObject.container.js]: Something like that ?

```javascript
const filtered = filterAll(children, query, { nameAttr }, `${currentPosition} > ${name}`);
results = results.concat(filtered);
```

File: packages/containers/examples/ExampleSelectObject.js [hunk 3]
```diff
@@ -50,5 +50,11 @@ const ExampleSelectObject = {
 			<SelectObject {...props} schema={schema} />
 		</div>
 	),
+	'tree with filter mode set to ALL': () => (
+		<div>
+			<IconsProvider />
+			<SelectObject {...props} filterMode={'ALL'} />
```
Comment (jmfrancois) [inline packages/containers/examples/ExampleSelectObject.js]: `<SelectObject {...props} filterMode={SelectObject.ALL} />`

**Pull request merged**